op-node/rollup/finality/finalizer.go

```diff
@@ -111,7 +111,7 @@ func (fi *Finalizer) FinalizedL1() (out eth.L1BlockRef) {
 }
 
 // Finalize applies a L1 finality signal, without any fork-choice or L2 state changes.
-func (fi *Finalizer) Finalize(ctx context.Context, l1Origin eth.L1BlockRef) {
+func (fi *Finalizer) Finalize(ctx context.Context, cfg *rollup.Config, l1Origin eth.L1BlockRef) {
```

Comment:
I am a bit concerned about passing around the cfg which resulted in many file changes.

instead, I feel we can store the cfg in the struct.

fortunately, I found out that the cfg is already passed into this function: 

```
func NewFinalizer(log log.Logger, cfg *rollup.Config, l1Fetcher FinalizerL1Interface, ec FinalizerEngine) *Finalizer {
	lookback := calcFinalityLookback(cfg)
	return &Finalizer{
		log:              log,
		finalizedL1:      eth.L1BlockRef{},
		triedFinalizeAt:  0,
		finalityData:     make([]FinalityData, 0, lookback),
		finalityLookback: lookback,
		l1Fetcher:        l1Fetcher,
		ec:               ec,
	}
}
```

so my suggestion is to add a new field in

```
type Finalizer struct {
	mu sync.Mutex

	log log.Logger

	// finalizedL1 is the currently perceived finalized L1 block.
	// This may be ahead of the current traversed origin when syncing.
	finalizedL1 eth.L1BlockRef

	// triedFinalizeAt tracks at which L1 block number we last tried to finalize during sync.
	triedFinalizeAt uint64

	// Tracks which L2 blocks where last derived from which L1 block. At most finalityLookback large.
	finalityData []FinalityData

	// Maximum amount of L2 blocks to store in finalityData.
	finalityLookback uint64

	l1Fetcher FinalizerL1Interface

	ec FinalizerEngine
}
```

Reply:
thanks - this is much much cleaner, have updated accordingly!